apps/teams-test-app/src/components/DialogAPIs.tsx

```diff
@@ -16,29 +16,56 @@ const DialogAPIs = (): ReactElement => {
             throw new Error('Url undefined');
           }
         },
-        submit: async (dialogInfo, setResult) => {
-          const onComplete = (err: string, result: string | object): void => {
-            setResult('Error: ' + err + '\nResult: ' + result);
-          };
-          // Store the reference of child window in React
-          const childWindow = dialog.open(dialogInfo, onComplete);
-          childWindow.addEventListener('message', (message: string) => {
-            // Message from parent
-            setResult(message);
-          });
-          childWindowRef.current = childWindow;
-          return '';
+        submit: {
+          withPromise: async (dialogInfo, setResult) => {
+            const onComplete = (err: string, result: string | object): void => {
+              setResult('Error: ' + err + '\nResult: ' + result);
+            };
+            // Store the reference of child window in React
+            const childWindow = dialog.open(dialogInfo, onComplete);
+            childWindow.addEventListener('message', (message: string) => {
+              // Message from parent
+              setResult(message);
+            });
+            childWindowRef.current = childWindow;
+            return '';
+          },
+          withCallback: (taskInfo, setResult) => {
+            const onComplete = (err: string, result: string | object): void => {
+              setResult('Error: ' + err + '\nResult: ' + result);
+            };
+            // Store the reference of child window in React
+            const childWindow = tasks.startTask(taskInfo, onComplete);
+            childWindow.addEventListener('message', (message: string) => {
+              // Message from parent
+              setResult(message);
+            });
+            childWindowRef.current = childWindow;
+          },
         },
       },
     });
 
   const ResizeDialog = (): ReactElement =>
-    ApiWithTextInput<DialogInfo>({
+    ApiWithTextInput<DialogInfo | TaskInfo>({
       name: 'dialogResize',
       title: 'Dialog Resize',
-      onClick: async dialogInfo => {
-        dialog.resize(dialogInfo);
-        return 'Teams client SDK call dialog.resize was called';
+      onClick: {
+        validateInput: input => {
+          if (input.height === undefined && input.width === undefined) {
+            throw new Error('Height and width undefined');
+          }
+        },
+        submit: {
+          withPromise: async dialogInfo => {
+            dialog.resize(dialogInfo);
+            return 'Teams client SDK call dialog.resize was called';
+          },
+          withCallback: (taskInfo, setResult) => {
+            tasks.updateTask(taskInfo);
+            setResult('Teams client SDK call tasks.updateTask was called');
```

Comment:
would the difference in the result string here mean we'll need separate tests for both promise and callback?

Reply:
Discussed offline.